ssz/sedes/serializable.py

```diff
@@ -0,0 +1,419 @@
+import abc
+import collections
+import copy
+import re
+
+from eth_utils import (
+    to_dict,
+    to_set,
+    to_tuple,
+)
+
+from ssz.constants import (
+    CONTAINER_PREFIX_LENGTH,
+)
+from ssz.exceptions import (
+    DeserializationError,
+    SerializationError,
+)
+
+
+class MetaBase:
+    fields = None
+    field_names = None
+    field_attrs = None
+
+
+def _get_duplicates(values):
+    counts = collections.Counter(values)
+    return tuple(
+        item
+        for item, num in counts.items()
+        if num > 1
+    )
+
+
+def validate_args_and_kwargs(args, kwargs, arg_names, allow_missing=False):
+    duplicate_arg_names = _get_duplicates(arg_names)
+    if duplicate_arg_names:
+        raise TypeError("Duplicate argument names: {0}".format(sorted(duplicate_arg_names)))
+
+    needed_kwargs = arg_names[len(args):]
```

Comment:
```suggestion
    needed_arg_names = set(arg_names[len(args):])
```
`needed_kwargs` is slightly misleading as `kwargs` are usually `dict`s. The `set` is not strictly necessary, but I'd expect it `needed_kwargs` to have the same type as `used_kwargs`.